src/Umbraco.Core/IO/MediaFileSystem.cs

```diff
@@ -15,58 +16,60 @@ public class MediaFileSystem : FileSystemWrapper
 	{
 	    private readonly IContentSection _contentConfig;
 
-	    public MediaFileSystem(IFileSystem wrapped)
+	    public MediaFileSystem(IFileSystem2 wrapped)
 			: this(wrapped, UmbracoConfig.For.UmbracoSettings().Content)
 		{
 		}
 
-        public MediaFileSystem(IFileSystem wrapped, IContentSection contentConfig) : base(wrapped)
+        public MediaFileSystem(IFileSystem2 wrapped, IContentSection contentConfig) : base(wrapped)
         {
             _contentConfig = contentConfig;
         }
 
-	    public string GetRelativePath(int propertyId, string fileName)
+        // none of the methods below are used in Core anymore
+
+        [Obsolete("This low-level method should NOT exist.")]

```

Comment:
`GetRelativePath(string fullPathOrUrl)` is part of `IFileSystem` but these two overloads are directly dealing with the actual media path, which we want to avoid, as we want to use the higher-level `MediaFileSystem` methods.